modules/tinymce/src/plugins/codesample/test/ts/webdriver/CodeSampleCopyAndPasteTest.ts

```diff
@@ -0,0 +1,137 @@
+import { ApproxStructure, Keys, RealClipboard, RealMouse } from '@ephox/agar';
+import { beforeEach, describe, it } from '@ephox/bedrock-client';
+import { PlatformDetection } from '@ephox/sand';
+import { TinyAssertions, TinyContentActions, TinyHooks, TinySelections, TinyUiActions } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/codesample/Plugin';
+
+import * as TestUtils from '../module/CodeSampleTestUtils';
+
+describe('webdriver.tinymce.plugins.codesample.CodeSampleCopyAndPasteTest', () => {
+  const hook = TinyHooks.bddSetup<Editor>({
+    plugins: 'codesample',
+    toolbar: 'codesample',
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin ]);
+
+  const pressEnter = (editor: Editor) => TinyContentActions.keystroke(editor, Keys.enter());
+
+  const browser = PlatformDetection.detect().browser;
```

Comment:
We normally try to keep variables and functions grouped, so maybe move this to the top above the hook declaration?

Reply:
but functions _are_ variables!

I nearly just inlined it, now that `PlatformDetection.detect()` is a thunk, but I just moved it for now.

Reply:
yes, but not from a use-ability PoV. It's just much easier to find things if they are grouped like this (similar to why we keep types at the top of a file).